test/appstats_test.js

```diff
@@ -251,6 +252,44 @@ describe('App Stats tests', function() {
             });
         });
     });
+
+    describe('AzureAppInsightStats test', function () {
+        beforeEach(function () {
+        });
+        afterEach(function (done) {
+            fs.unlink(mock.AL_TOKEN_CACHE_FILENAME, function (err) {
+                done();
+            });
+            nock.cleanAll();
+        });
+
+        it('checks getAppStats() with no Functions', function (done) {
+            var expectedStats = {
+                statistics: []
+            };
+            var stats = new AzureAppInsightStats();
+            stats.getAppStats('2022-12-22T14:31:39', function (err, appStats) {
+              
+                assert.deepEqual(expectedStats, appStats);
+                done();
+            });
+        });
+
+        it('checks getAppStats() with empty or zero stats', function (done) {
+            var expectedStats = {
+                statistics: [
+                    { Master: { invocations: 0, errors: 0 } },
+                    { Collector: { invocations: 0, errors: 0 } },
+                    { Updater: { invocations: 0, errors: 0 } }
+                ]
+            };
+            var stats = new AzureAppInsightStats(DEFAULT_APP_FUNCTIONS);
+            stats.getAppStats('2022-12-22T14:31:39', function (err, appStats) {
+                assert.deepEqual(expectedStats, appStats);
+                done();
+            });
+        });
+    });
     
```

Comment:
These tests are good.
But could you please add one more which checks `AlAzureMaster. getStats()`